go/enclave/components/rollup_producer.go

```diff
@@ -0,0 +1,155 @@
+package components
+
+// todo (@stefan) - once the cross chain messages based bridge is implemented remove this completely
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	gethlog "github.com/ethereum/go-ethereum/log"
+
+	"github.com/obscuronet/go-obscuro/contracts/generated/MessageBus"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/errutil"
+
+	"github.com/obscuronet/go-obscuro/go/enclave/crypto"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/obscuronet/go-obscuro/go/enclave/core"
+	"github.com/obscuronet/go-obscuro/go/enclave/db"
+)
+
+// rollupProducer encapsulates the logic of decoding rollup transactions submitted to the L1 and resolving them
+// to rollups that the enclave can process.
+type rollupProducer struct {
+	// TransactionBlobCrypto- This contains the required properties to encrypt rollups.
+	TransactionBlobCrypto crypto.TransactionBlobCrypto
+
+	ObscuroChainID  int64
+	EthereumChainID int64
+
+	logger gethlog.Logger
+
+	storage db.Storage
+
+	registry      BatchRegistry
```

Comment:
the naming of this variable is inconsistent in the places where it's used.
Sometimes `registry` or `batchRegistry`.
Let's settle on `batchRegistry` everywhere